app/src/main/java/com/mensinator/app/CalendarScreen.kt

```diff
@@ -1,211 +1,105 @@
 package com.mensinator.app
 
-
 import android.content.Context
 import android.util.Log
 import android.widget.Toast
-import androidx.compose.foundation.background
-import androidx.compose.foundation.border
-import androidx.compose.foundation.clickable
-import androidx.compose.foundation.gestures.detectHorizontalDragGestures
 import androidx.compose.foundation.*
 import androidx.compose.foundation.layout.*
-import androidx.compose.foundation.shape.CircleShape
-import androidx.compose.material.icons.Icons
-import androidx.compose.material.icons.automirrored.filled.ArrowBack
-import androidx.compose.material.icons.automirrored.filled.ArrowForward
-import androidx.compose.ui.res.painterResource
 import androidx.compose.material3.Button
-import androidx.compose.material3.Icon
-import androidx.compose.material3.IconButton
+import androidx.compose.material3.MaterialTheme
 import androidx.compose.material3.Text
 import androidx.compose.runtime.*
 import androidx.compose.ui.Alignment
 import androidx.compose.ui.Modifier
-import androidx.compose.ui.draw.drawWithContent
-import androidx.compose.ui.geometry.Offset
 import androidx.compose.ui.graphics.Color
-import androidx.compose.ui.input.pointer.pointerInput
+import androidx.compose.ui.text.style.TextAlign
+import androidx.compose.ui.unit.dp
+import com.kizitonwose.calendar.compose.VerticalCalendar
+import com.kizitonwose.calendar.compose.rememberCalendarState
+import com.mensinator.app.navigation.displayCutoutExcludingStatusBarsPadding
+import java.time.DayOfWeek
+import java.time.LocalDate
+import java.time.YearMonth
+import com.kizitonwose.calendar.core.*
+import com.mensinator.app.data.ColorSource
+import com.mensinator.app.ui.theme.isDarkMode
+import org.koin.compose.koinInject
+import androidx.compose.foundation.shape.CircleShape
+import androidx.compose.material3.HorizontalDivider
 import androidx.compose.ui.platform.LocalContext
 import androidx.compose.ui.res.stringResource
 import androidx.compose.ui.text.font.FontWeight
-import androidx.compose.ui.text.style.TextAlign
-import androidx.compose.ui.unit.dp
 import androidx.compose.ui.unit.sp
-import com.mensinator.app.data.ColorSource
-import com.mensinator.app.navigation.displayCutoutExcludingStatusBarsPadding
 import com.mensinator.app.settings.ResourceMapper
 import com.mensinator.app.settings.StringSetting
-import com.mensinator.app.ui.theme.isDarkMode
-import org.koin.compose.koinInject
-import java.time.DayOfWeek
-import java.time.LocalDate
 import java.time.temporal.ChronoUnit
-import kotlin.math.abs
+import java.util.Calendar
+import java.util.Locale
 
 /*
-This file creates the calendar. A sort of "main screen".
+This function is the initiator of the vertical calendar.
  */
 @Composable
 fun CalendarScreen(modifier: Modifier) {
```

Comment:
We must not use Set/Map/List directly anymore in composables. Using these makes the composables not skip, so on every scroll, everything gets redrawn which causes bad performance. Probably https://github.com/Kotlin/kotlinx.collections.immutable should be used, I use this at work.